tests/tests_lite/strategies/test_xla.py

```diff
@@ -113,3 +114,17 @@ def test_xla_validate_unsupported_iterable_dataloaders(_, dataloader, monkeypatc
 
     with pytest.raises(TypeError, match="TPUs do not currently support"):
         XLAStrategy().process_dataloader(dataloader)
+
+
+def tpu_all_gather_fn(strategy):
+    for sync_grads in [True, False]:
+        tensor = torch.tensor(1, device=strategy.root_device)
+        result = strategy.all_gather(tensor, sync_grads=sync_grads)
+        assert result.sum() == 8
```

Comment:
How about we do a backward and show gradients gets synced?

```suggestion
    tensor = torch.tensor(1, device=strategy.root_device)
    result = strategy.all_gather(tensor, sync_grads=False)
    assert result.sum() == 8
    
    tensor = torch.tensor(1.0, device=strategy.root_device , requires_grad=True)
    result = strategy.all_gather(tensor, sync_grads=False)
    result.sum().backward()
    assert torch.equal(tensor.grad, torch.tensor(1.0))
```

Reply:
Good idea, I totally agree, this test is a lot more meaningful.

Reply:
The TPU tests are failing, it was already the case before the last changes. Do you know how I can access the full output logs of the test? I only see:

```
Waiting for job to finish: ..........................................................Failed:1
Done waiting. Job status code: 1
Error: Process completed with exit code 1.
```

You got a suggestion how to run the TPU tests before committing changes? Maybe you already got a Colab notebook ready somewhere? 😁